### ISSUE TITLE
Nikon Z6III

### ISSUE THREAD

**Issue by rick05chap** [darktable-org/darktable#17075]
Uploaded the first image. Will upload the remaining images in the next couple of minutes.

Nikon Z6 _3 Lossless compressed FX.NEF
Nikon Z6 _3 High Efficiency 1_1.NEF
Nikon Z6 _3 High Efficiency 16_9.NEF
Nikon Z6 _3 High Efficiency DX.NEF
Nikon Z6 _3 High Efficiency FX.NEF
Nikon Z6 _3 High Efficiency Star 1_1.NEF
Nikon Z6 _3 High Efficiency Star 16_9.NEF
Nikon Z6 _3 High Efficiency Star DX.NEF
Nikon Z6 _3 High Efficiency Star FX.NEF
Nikon Z6 _3 Lossless compressed 1_1.NEF
Nikon Z6 _3 Lossless compressed 16_9.NEF
Nikon Z6 _3 Lossless compressed DX.NEF

**Comment by LebedevRI** [comment]
Lovely.
@rick05chap thank you for contributing full sample set!

**Comment by rick05chap** [comment]
Roman,Please let me know if there is anything else I can submit that would be of use.Rick ChapmanOn Jun 29, 2024, at 5:07 PM, Roman Lebedev ***@***.***> wrote:﻿
Lovely.
@rick05chap thank you for contributing full sample set!

**Comment by Dantoes** [comment]
Hi, I was wondering if there are any updates on this. I'd like to be able to edit the photos I've taken with this camera. Thanks!

**Comment by igillingham** [comment]
Hello - just checking to see if there's any progress for Z 6 3 support? I love Darktable but have not been able to use it since acquiring my new camera. I am a software developer so could possibly help out if needed.

**Comment by Dantoes** [comment]
I added your changes to `/usr/share/darktable/rawspeed/cameras.xml`, started darktable 4.8.1, removed the film roll containing the  Z6_3 images, deleted all *.xmp files in that directory and then re-imported the directory. When I try to open a photo it fails with the following log messages:

```
    51,3639 [rawspeed] (DSC_2878.NEF) rawspeed::RawImage rawspeed::RawDecoder::decodeRaw(), line 338: rawspeed::Buffer::size_type rawspeed::ByteStream::check(rawspeed::Buffer::size_type) const, line 64: Out of bounds access in ByteStream
    51,3639 [libraw_open] extensions whitelist: `cr3'
    51,3640 [qoi_open] no proper file header in <path>/DSC_2878.NEF
    51,3652 [temperature] failed to read camera white balance information from `DSC_2878.NEF'!
    51,3652 [temperature] failed to read camera white balance information from `DSC_2878.NEF'!
    51,5982 [temperature] failed to read camera white balance information from `DSC_2878.NEF'!
    51,5982 [temperature] `Nikon Z6_3' color matrix not found for image
    51,5983 [temperature] failed to read camera white balance information from `DSC_2878.NEF'!
```

Do I have to compile the develop-branch with your changes applied? I figured it shouldn't be necessary, since it's a just config file change.

**Comment by Dantoes** [comment]
You are right, the images are HE-Star-compressed. I thought I had changed the setting, but it looks like I didn’t. Thank you for the hint!

**Comment by Fade78** [comment]
Thanks, i'll take a look this evening. I meant lossless compression so it's perfect.

**Comment by Fade78** [comment]
It works! I modified /var/lib/flatpak/app/org.darktable.Darktable/x86_64/stable/(some uuid here)/files/share/darktable/rawspeed/cameras.xml

**Comment by Fade78** [comment]
So the "Lens correction" module display an error (body not found) - I replaced with Z6 II and it applied some correction, I hope this is not to far from truth. That's very minor, I'm very happy to be able to use my RAWs before the official release.

**Comment by Fade78** [comment]
It works also. The file is /var/lib/flatpak/app/org.darktable.Darktable/x86_64/stable/(some uuid)/files/share/lensfun/version_1/mil-nikon.xml

Today feels like Christmas.

**Comment by LebedevRI** [comment]
@rick05ch Hello. RPU is finally up. Please contribute the full sample set.

**Comment by LebedevRI** [comment]
@rick05chap  Hello. RPU is finally up. Please contribute the full sample set.

**Comment by LebedevRI** [comment]
poke @rick05chap @dcheleme

**Comment by LebedevRI** [comment]
poke @rick05chap @dchelemen

**Comment by Fade78** [comment]
> [@FTouzain](https://github.com/FTouzain) [@Fade78](https://github.com/Fade78) [@Dantoes](https://github.com/Dantoes) [@igillingham](https://github.com/igillingham) Unfortunately samples were lost when RPU crashed last summer. Please help by submitting a _full_ new set.

Sorry, it was a long time. I'm not sure of what you ask here. The white balance data I've made for the Z6iii ?

**Comment by LebedevRI** [comment]
@Fade78 Please disregard, the question i had has been resolved.
